lib/spack/spack/test/views.py

```diff
@@ -23,3 +24,55 @@ def test_remove_extensions_ordered(install_mockery, mock_fetch, tmpdir):
 
     e1 = e2["extension1"]
     view.remove_specs(e1, e2)
+
+
+@pytest.mark.regression("32456")
+def test_view_with_spec_not_contributing_files(install_mockery, mock_fetch, tmpdir):
+    # This test is a bit contrived, but it checks whether
+    # pkg.add_files_to_view() receives the correct source paths
+    # when another package does not have any files to
+    # contribute to a view. That can happen either because
+    # there simply are no files in the prefix, or because all
+    # its files conflict with another spec -- here we test the
+    # former).
+
+    view_dir = str(tmpdir)
+    layout = DirectoryLayout(view_dir)
+    view = SimpleFilesystemView(view_dir, layout)
+
+    a = Spec("libelf").concretized()
+    b = Spec("trivial-install-test-package").concretized()
+    a.package.do_install()
+    b.package.do_install()
+
+    # Remove all files of pkg a
+    for root, _, files in os.walk(a.prefix):
+        for name in files:
+            os.remove(os.path.join(root, name))
+
+    # Create some dirs
+    os.makedirs(a.prefix.first)
+    os.makedirs(a.prefix.second)
+    os.makedirs(b.prefix.second)
+    os.makedirs(b.prefix.third)
```

Comment:
Making the directories for `b` is confusing since they add nothing to the merge map (the above suggestion to fully control the installation prefix within this test would help with that I think).

Reply:
done